### PR TITLE
Unreserve dispute bonds when destroying a market

### LINKED ISSUE #686: [prediction-markets] `admin_destroy_market` doesn't clear dispute bonds
It is possible to destroy a market in `Disputed` state using `admin_destroy_market`, but this will not result in the dispute bonds being released.

This is a bug, but it would be very unusual for a market to be deleted when it is already in `Disputed` state. So I'm putting low priority.

### PR DESCRIPTION
**Pull request by maltekliemann**
Fixes #686. Strictly speaking, we may have some markets on Battery Station which were destroyed in disputed state (haven't checked yet), so this would require a migration. But since this storage isn't really critical, we'll just leave it there. If it ever bothers us, we can remove it using governance.

### CODE REVIEW

**Comment by maltekliemann** [comment]
Note that this doesn't require changes to the benchmarks, since we're already considering the amount of disputes.

File: zrml/prediction-markets/src/lib.rs [hunk 1]
```diff
@@ -168,6 +168,18 @@ mod pallet {
...
             let (ids_len, disputes_len) = Self::clear_auto_resolve(&market_id)?;
+            // `Disputes` is emtpy unless the market is disputed, so this is just a defensive
+            // check.
+            if market.status == MarketStatus::Disputed {
+                for (index, dispute) in Disputes::<T>::take(market_id).iter().enumerate() {
+                    T::AssetManager::unreserve_named(
+                        &Self::reserve_id(),
+                        Asset::Ztg,
+                        &dispute.by,
+                        default_dispute_bond::<T>(index),
```
Comment (sea212) [inline zrml/prediction-markets/src/lib.rs:179]: Do we already track storing the bonded amount (In case the dispute amount changes)?

**Review by sea212 — APPROVED** [review]
LGTM

File: zrml/prediction-markets/src/lib.rs (hunk 1 above)
Comment (maltekliemann) [reply zrml/prediction-markets/src/lib.rs:179]: No, that's somewhere in ZIP-0, I believe.

File: zrml/prediction-markets/src/tests.rs [hunk 2]
```diff
@@ -427,6 +428,64 @@ fn admin_destroy_market_correctly_slashes_permissionless_market_disputed() {
...
+                account,
+                SENTINEL_AMOUNT,
+            ));
+        };
+        set_up_account(&CHARLIE);
+        set_up_account(&DAVE);
+
+        let balance_free_before_charlie = Balances::free_balance(&CHARLIE);
+        let balance_free_before_dave = Balances::free_balance(&DAVE);
+        assert_ok!(PredictionMarkets::admin_destroy_market(Origin::signed(SUDO), 0));
```
Comment (Chralt98) [inline zrml/prediction-markets/src/tests.rs:469]: A check that `Disputes` at market id 0 is actually empty could be useful here. But not so important after ZIP-0.

**Review by Chralt98 — DISMISSED** [review]

File: zrml/prediction-markets/src/tests.rs (hunk 2 above)
Comment (maltekliemann) [reply zrml/prediction-markets/src/tests.rs]: [f6bbfb3](https://github.com/zeitgeistpm/zeitgeist/pull/877/commits/f6bbfb3a3371d3b0774f01b5d49763d842433c34)

**Comment by Chralt98** [comment]
Starting my review now..

**Review by Chralt98 — APPROVED** [review]
OK, although I am in favour of completely moving the dispute bond to `simple-disputes`, if not remove `simple-disputes` completely.

**Comment by maltekliemann** [comment]
> OK, although I am in favour of moving the dispute bond to `simple-disputes`, if not remove `simple-disputes` completely.

Sure, but right now, we're still using the old system.

**Pull request merged**